signals/signal.go

```diff
@@ -41,3 +43,33 @@ func SetupSignalHandler() (stopCh <-chan struct{}) {
 
 	return stop
 }
+
+// NewContext creates a new context with SetupSignalHandler()
+// as our Done() channel.
+func NewContext() context.Context {
+	return &signalContext{stopCh: SetupSignalHandler()}
+}
+
+type signalContext struct {
+	stopCh <-chan struct{}
+}
+
+// Deadline implements context.Context
+func (scc *signalContext) Deadline() (deadline time.Time, ok bool) {
+	return time.Now(), false
+}
+
+// Done implements context.Context
+func (scc *signalContext) Done() <-chan struct{} {
+	return scc.stopCh
+}
+
+// Err implements context.Context
+func (scc *signalContext) Err() error {
```

Comment:
Probably should follow the semantics of `Err()`as described in the `Context` [godoc](https://golang.org/pkg/context/#Context)

```
If Done is closed, Err returns a non-nil error explaining why:
```

Reply:
Done.